src/rgw/rgw_cr_rados.cc

```diff
@@ -837,7 +837,7 @@ int RGWSyncLogTrimCR::request_complete()
     return r;
   }
   // nothing left to trim, update last_trim_marker
-  if (*last_trim_marker < to_marker) {
+  if (*last_trim_marker < to_marker && to_marker != max_marker) {
```

Comment:
I'm trying to understand this logic in relation to the commit text -- "if we set last_trim=99999999, it will never attempt to trim again". It looks like this logic prevents last_trim_marker from being set to max_marker.

Reply:
sorry if the commit message wasn't clear

this `last_trim_marker` stuff is part of an optimization in mdlog/datalog trimming that avoids resending trim requests for markers that we know we've already trimmed in a previous trim interval. when `RGWSyncLogTrimCR` gets `ENODATA` back, that indicates the log is fully trimmed up to the given `to_marker`. by recording that value in `last_trim_marker`, `DataLogTrimCR::operate()` will avoid sending any more requests until it sees a higher marker

but in this special case of max_marker, we'll never see a higher marker. if we did record this value in `last_trim_marker`, that would prevent `DataLogTrimCR` from ever sending any more requests on that shard, even though more entries may have been written

so the intended behavior here is to reissue a trim request at `max_marker` every trim interval to keep clearing the log of new events